source/parse.h

```diff
@@ -2244,6 +2244,7 @@ struct declaration_node
     capture_group captures;
 
     source_position pos;
+    source_position location; // Actual source location, accounting for explicit, non-default access specifier.
```

Comment:
A declarations starts at its _access-specifier_ if it has one.
But `declaration_node::position` currently returns the position of its identifier.
And `declaration_node::pos` doesn't hold a source position, but a stream position.
So I added `declaration_node::location` to save the actual source location.
This Cpp2 fragment:
```Cpp2
  a: int         = 0;
  public b: int  = 0;
  private c: int = 0;
```
Now has this lowered diff:
```diff
   private: using y_a_as_base::a;
-         public: using y_b_as_base::b;
+  public: using y_b_as_base::b;
   private: using y_c_as_base::c;
```